### PR TITLE
3875 Avoid indexing non-RECAP Dockets into ES.

### PR DESCRIPTION
**Pull request by albertisfu**
According to #3875, this PR prevents indexing signals from indexing non-RECAP Dockets into ES.

**It considers the following:**

- New dockets, whose sources do not include RECAP, are not indexed into ES.
- Dockets that are updated and whose source does not include RECAP, are not indexed into ES.
- If a non-RECAP Docket's `source` is updated to include a RECAP source (e.g: receives an upload from RECAP), the docket is indexed into ES.
- It avoids calling unnecessary celery tasks for non-RECAP dockets since the condition is checked before calling a celery task.
- The `"source"` field is tracked but is not indexed into RECAP documents in ES.
- This condition does not apply to other indices, where non-RECAP dockets are still tracked to spread their changes to the main ES documents, like in Opinions and OA.
- New tests have been added, and existing ones were updated according to the adjustments.

**The logic assumes the following:**
- A docket that receives docket entries belongs to the RECAP collection, so filings are always indexed without checking if the Docket belongs to a RECAP source.
- A docket that gets a RECAP source cannot revert to a non-RECAP source.

If any of these assumptions can be false under certain conditions, please let me know since it will require additional logic to prevent indexing filings and to remove content when it's not a RECAP source anymore.

### CODE REVIEW

File: cl/lib/es_signal_processor.py [hunk 1]
```diff
@@ -741,6 +741,15 @@ def handle_save(
             return None
 
         mapping_fields = self.documents_model_mapping["save"][sender]
+        if (
+            isinstance(instance, Docket)
+            and not instance.source in Docket.RECAP_SOURCES
+            and mapping_fields.get("self", None)
```
Comment (mlissner) [inline cl/lib/es_signal_processor.py:747]: What does this line do? Worth adding a comment?

**Review by mlissner — APPROVED** [review]
LGTM. One little suggestion and then let's have Eduardo review. Once you're both happy, let's merge.

File: cl/lib/es_signal_processor.py (hunk 1 above)
Comment (albertisfu) [reply cl/lib/es_signal_processor.py]: Sure! That line filter out signals that doesn't affect DocketDocuments in ES.
I've added a comment.

**Review by ERosendo — APPROVED** [review]
The code looks good 👍

**Comment by mlissner** [comment]
Rebased again and set for auto-merge. Thanks all!

**Pull request merged**